e621/endpoints/my_endpoints.py

```diff
@@ -1,18 +1,239 @@
+                "search[is_active]": is_active,
+                "limit": limit,
+            },
+        ).json()
+        return [Note(**note) for note in raw_notes] if isinstance(raw_notes, list) else []
+
+    def create(self, post_id: int, x: int, y: int, width: int, height: int, body: str) -> Note:
+        return Note(
+            **self._api.session.post(
+                "notes",
+                params={
+                    "note[post_id]": post_id,
+                    "note[x]": x,
+                    "note[y]": y,
+                    "note[width]": width,
+                    "note[height]": height,
+                    "note[body]": body,
+                },
+            ).json()
+        )
+
+    def update(self, note_id: int, post_id: int, x: int, y: int, width: int, height: int, body: str) -> Note:
+        return Note(
+            **self._api.session.put(
+                f"notes/{note_id}",
+                params={
+                    "note[post_id]": post_id,
+                    "note[x]": x,
+                    "note[y]": y,
+                    "note[width]": width,
+                    "note[height]": height,
+                    "note[body]": body,
+                },
+            ).json()
+        )
+
+    def delete(self, note_id: int) -> None:
+        self._api.session.delete("notes/{note_id}")
+
+    def revert(self, note_id: int, version_id: int) -> None:
+        self._api.session.put(f"notes/{note_id}/revert", params={"version_id": version_id})
 
 
 class Pools(BaseEndpoint):
-    pass
+    def get(self, pool_id: int) -> EnrichedPool:
+        return EnrichedPool(**self._api.session.get(f"pools/{pool_id}").json())
+
+    def search(
+        self,
+        name_matches: Optional[str] = None,
+        id: Union[int, List[int], None] = None,
+        description_matches: Optional[str] = None,
+        creator_name: Optional[str] = None,
+        creator_id: Optional[int] = None,
+        is_active: Optional[bool] = None,
+        is_deleted: Optional[bool] = None,
+        category: Optional[PoolCategory] = None,
+        order: Optional[Literal["name", "created_at", "updated_at", "post_count"]] = None,
+        limit: Optional[int] = None,
+    ) -> List[EnrichedPool]:
+        raw_pools = self._api.session.get(
+            "pools",
+            params={
+                "search[name_matches]": name_matches,
+                "search[id]": ",".join(map(str, id)) if isinstance(id, list) else id,
+                "search[description_matches]": description_matches,
+                "search[creator_name]": creator_name,
+                "search[creator_id]": creator_id,
+                "search[is_active]": is_active,
+                "search[is_deleted]": is_deleted,
+                "search[category]": category,
+                "search[order]": order,
+                "limit": limit,
+            },
+        ).json()
+        return [EnrichedPool(**pool) for pool in raw_pools]

```

Comment:
You're not handling the empty case again. If you want to, we can skip this for now and just switch to my implementation after we merge your pull request. This way we won't need to handle empty cases because they are handled by the base class.